lib/models/ObjectMDArchive.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * class representing the archive of object metadata.
+ *
+ * @class
+ */
+class ObjectMDArchive {
+    /**
+     *
+     * @constructor
+     * @param {Object} archiveInfo contains the archive info set by the TLP and returned by the TLP jobs
+     * @param {Date} [restoreRequestedAt] set at the time restore request is made by the client
+     * @param {Number} [restoreRequestedDays] set at the time restore request is made by the client
+     * @param {Date} [restoreCompletedAt] set at the time of successful restore
+     * @param {Date} [restoreWillExpireAt] computed and stored at the time of restore
+     * @throws {Error} case of invalid parameter
+     */
+    constructor(
+        archiveInfo,
+        restoreRequestedAt = undefined,
+        restoreRequestedDays = undefined,
+        restoreCompletedAt = undefined,
+        restoreWillExpireAt = undefined) {
+        this.setArchiveInfo(archiveInfo);
+        this.setRestoreRequestedAt(restoreRequestedAt);
+        this.setRestoreRequestedDays(restoreRequestedDays);
+        this.setRestoreCompletedAt(restoreCompletedAt);
+        this.setRestoreWillExpireAt(restoreWillExpireAt);
+    }
+
+    /**
+     *
+     * @param {Object} data archiveInfo
+     * @returns {boolean} true if the provided object is valid
+     */
+    static isValid(data) {
+        try {
+            const archiveMD = new ObjectMDArchive({});
+            archiveMD.setArchiveInfo(data.archiveInfo);
+            archiveMD.setRestoreRequestedAt(data.restoreRequestedAt);
+            archiveMD.setRestoreRequestedDays(data.restoreRequestedDays);
+            archiveMD.setRestoreCompletedAt(data.restoreCompletedAt);
+            archiveMD.setRestoreWillExpireAt(data.restoreWillExpireAt);
+            return true;
+        } catch (err) {
+            return false;
+        }
+    }
```

Comment:
The reason why we have this function is that the current Arsenal doesn't accept class objects in the MD in the tests, but at the same time, we want to make sure the provided object is correct. So I introduce a "isValid" function in both new models to make sure, when we use the setter, that the provided object is either of the right class or is valid, using the model function. It allows to pass the unit tests.